### PR TITLE
improveExportsource and enable simplified ribbon handling.

### PR DESCRIPTION
**Pull request by hecon5**
For #157, this changes export behavior and autocloses form; also adds in ability to go right to settings from the "ribbon".

### CODE REVIEW

**Comment by joyfullservice** [comment]
Sorry for the conflicts on `modInstall.bas`... I think we were both working on some similar things.  😄  I think I have finished the parts I was wanting to implement on the installation process, so feel free to update the PR with the latest changes.

File: Version Control.accda.src/modules/modAddIn.bas [hunk 1]
```diff
@@ -15,6 +15,33 @@ Public Enum eReleaseType
     Same_Version = 3
 End Enum
 
+Public Enum RibbonLaunch
```
Comment (joyfullservice) [inline Version Control.accda.src/modules/modAddIn.bas:18]: Could we prefix these with the standard convention used in the rest of the project?
```
Public Enum eRibbonLaunch
    erlVCSOpen
    erlVCSOptions
```
etc...

File: Version Control.accda.src/modules/modAddIn.bas [hunk 2]
```diff
@@ -15,6 +15,33 @@ Public Enum eReleaseType
...
+    
+    VCSOpen = 0
```
Comment (joyfullservice) [inline Version Control.accda.src/modules/modAddIn.bas:20]: If these are sequential, we don't even need the assignments. (We can/should always refer to them by name anyway.) Just helps simplify readability. 😄

File: Version Control.accda.src/modules/modAddIn.bas [hunk 3]
```diff
@@ -15,6 +15,33 @@ Public Enum eReleaseType
...
+' Author    : hecon5
+' Date      : 2/05/2020
+' Purpose   : Launch the main add-in form.
+'---------------------------------------------------------------------------------------
+'
+Private Function AddInLaunch(RibbonCmdIn as Long) As Boolean
+    PreloadVBE
+    Form_frmVCSMain.HandleCmd RibbonCmdIn
+    AddInLaunch = True
+End Function
```
Comment (joyfullservice) [inline Version Control.accda.src/modules/modAddIn.bas:44]: Nice! I like this approach.

File: Version Control.accda.src/modules/modInstall.bas [hunk 4]
```diff
@@ -121,9 +121,16 @@ Public Function InstallVCSAddin() As Boolean
         On Error GoTo 0
     Else
         On Error GoTo 0
+
+        ' Remove any Legacy Menu controls
```
Comment (joyfullservice) [inline Version Control.accda.src/modules/modInstall.bas:125]: We can move this to `CheckForLegacyInstall`.

File: Version Control.accda.src/modules/modAddIn.bas (hunk 1 above)
Comment (hecon5) [reply Version Control.accda.src/modules/modAddIn.bas:18]: yup; will update.

File: Version Control.accda.src/modules/modAddIn.bas (hunk 2 above)
Comment (hecon5) [reply Version Control.accda.src/modules/modAddIn.bas:20]: good point; will remove.

File: Version Control.accda.src/modules/modAddIn.bas (hunk 3 above)
Comment (hecon5) [reply Version Control.accda.src/modules/modAddIn.bas:44]: Thanks!

File: Version Control.accda.src/modules/modInstall.bas (hunk 4 above)
Comment (hecon5) [reply Version Control.accda.src/modules/modInstall.bas:125]: done

**Comment by hecon5** [comment]
ok; this runs for me; the  only thing I'd suggest is that the "open File" be defaulted to true, other than that, this works for me.

**Comment by joyfullservice** [comment]
> 
> 
> ok; this runs for me; the only thing I'd suggest is that the "open File" be defaulted to true, other than that, this works for me.

I think I would like to keep the defaults they way they are presently, since that will be what the majority of users will use. (Most would not need to trust the document each time, since the folder is already trusted.)

That being said, you are welcome to add a PR to save the selected install options to the registry, and default to the saved options when the form opens. That way you only have to check the box once, and it will be automatically checked for future updates. You can just follow the conventions used with `InstalledVersion`.

**Pull request merged**